src/dumbo/src/ns.rs

```diff
@@ -104,32 +109,40 @@ impl MmdsNetworkStack {
     // This is the entry point into the MMDS network stack. The src slice should hold the contents
     // of an Ethernet frame (of that exact size, without the CRC).
     pub fn detour_frame(&mut self, src: &[u8]) -> bool {
-        // The frame cannot possibly contain an ARP request or IPv4 packet for the MMDS.
-        if !test_speculative_tpa(src, self.ipv4_addr)
-            && !test_speculative_dst_addr(src, self.ipv4_addr)
-        {
+        let mut arp_reply_ipv4_addr = None;
+        for ipv4_addr in self.ipv4_addr_pool.iter() {
+            // Check if the frame contains an ARP frame or an IPv4 packet sent to one
+            // of the MMDS server IPv4 addresses.
+            if test_speculative_tpa(src, *ipv4_addr) || test_speculative_dst_addr(src, *ipv4_addr) {
+                arp_reply_ipv4_addr = Some(*ipv4_addr);
+            }
+        }
+
+        if arp_reply_ipv4_addr.is_none() {
             return false;
         }
 
         if let Ok(eth) = EthernetFrame::from_bytes(src) {
             match eth.ethertype() {
-                ETHERTYPE_ARP => return self.detour_arp(eth),
+                ETHERTYPE_ARP => return self.detour_arp(eth, arp_reply_ipv4_addr.unwrap()),
```

Comment:
I know it's pretty obvious that this `unwrap()` is safe but can we also add a comment about this?